File: 황영훈/week09/짝지어_제거하기.java
import java.util.Stack;

class Solution {
    public int solution(String s) {
        // 문자 하나씩 검사하기 위해 문자열 배열 생성
        String[] strArray = s.split("");
        // stack을 이용해 문제 해결
        Stack<String> stack = new Stack<>();
        // 문자 하나씩 검사
        for (int i = 0; i < s.length(); i++) {
            // 빈 스택이면 바로 push
            if (stack.isEmpty()) {
                stack.push(strArray[i]);
                continue;
            }
            // 두 문자가 같으면 pop, 다르면 push
            if (stack.peek().equals(strArray[i])) {
                stack.pop();
            } else {
                stack.push(strArray[i]);
            }
        }
        // 문자를 모두 검사한 후 stack이 비었으면 짝지어 제거하기 성공
        if (stack.isEmpty()) {
            return 1;
        }
        return 0;
    }
}

File: 이수빈/week9/짝지어제거하기.py
def solution(s):
    answer = 1
    st = [] # stack
    
    for i in range(len(s)):
        if len(st) and st[-1] == s[i]: # 마지막 원소가 s[i]인 경우 마지막 원소 제거
            st.pop()
        else:
            st.append(s[i]) # 그렇지 않은 경우 스택에 원소 삽입
            
    if len(st): # 스택에 원소가 남아있으면 짝지어 제거하기 실패
        answer = 0
    return answer